core/trino-main/src/main/java/io/trino/connector/informationschema/InformationSchemaMetadata.java

```diff
@@ -318,6 +319,7 @@ private Set<QualifiedTablePrefix> calculatePrefixesWithTableName(
                         metadata.listViews(session, prefix).stream()))
                 .filter(objectName -> predicate.get().test(asFixedValues(objectName)))
                 .map(QualifiedObjectName::asQualifiedTablePrefix)
+                .limit(MAX_PREFIXES_COUNT + 1)
```

Comment:
@findepi 
It is reasonable to use `limit` before collecting to set here as `listTables` and `listViews` will not return duplicates.